ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -211,25 +228,50 @@ final class ApiSubmissionService private (
           contractStore.lookupContractKey(commands.submitter, _),
           commands.commands
         )
-      submissionResult <- handleResult(res)
+      transactionInfo <- res.fold(error => {
+        Metrics.failedInterpretationsMeter.mark()
+        Future.failed(grpcError(toStatus(error)))
+      }, Future.successful)
+      partyAllocationResults <- if (configuration.implicitPartyAllocation) {
```

Comment:
If you could move nested logic into their own method I believe the readability would benefit.

Reply:
Done.